src/shapes/paint/linear_gradient.cpp

```diff
@@ -75,25 +74,29 @@ void LinearGradient::update(ComponentDirt value) {
             // Get the start and end of the gradient in world coordinates (world
             // transform of the shape).
             const Mat2D& world = m_ShapePaintContainer->worldTransform();
-            Vec2D worldStart = Vec2D::transform(start, world);
-            Vec2D worldEnd = Vec2D::transform(end, world);
-            makeGradient(worldStart, worldEnd);
-        } else {
-            makeGradient(start, end);
+            start = Vec2D::transform(start, world);
+            end = Vec2D::transform(end, world);
         }
+
         // build up the color and positions lists
-        double ro = opacity() * renderOpacity();
-        for (auto stop : m_Stops) {
-            paint->addStop(
-                colorModulateOpacity((unsigned int)stop->colorValue(), ro),
-                stop->position());
+        const double ro = opacity() * renderOpacity();
+        const auto count = m_Stops.size();
+        ColorInt colors[count];
```

Comment:
We were using variable length arrays (non compile time constant) somewhere else and someone from the community had trouble getting the Microsoft compiler to use them. We never validated it ourselves, but some brief searching suggests it's still an issue: https://developercommunity.visualstudio.com/t/about-arrys/402728

We should fire up windows and try with MSBuild.

Reply:
They're certainly easy to type :) but skia never relied on this (just fyi) -- we wrote a simple / resuable class for this sort of thing, which would prealloc N slots on the stack, and only fall back to a malloc if needed.